src/main/util/CartesianProductPartial.java

```diff
@@ -0,0 +1,22 @@
+package util;
+
+import primitives.Triple;
+
+abstract class CartesianProductPartial {
+
+	public double xx;
+	public double yy;
+	public double zz;
+	public double yx;
```

Comment:
```suggestion
	public double xy;
```
name the base class fields in alphabetical order, and the child can add the backwards ones

Reply:
oyoyoyoy. That's a real headache

Reply:
Then I have to reorder all this math in the intersect function.

Reply:
You can just rename the variables and it should update everything. (Use a temporary, as in the swap algorithm you learned in first year)